### PR TITLE
Chore: contracts/ directory housekeeping

### PR DESCRIPTION
**Pull request by alexroan**
Changes:
- In the `contracts/` directory, change the path from `contracts/contracts/v0.8/...` to `contracts/src/v0.8/...` to follow `evm-contracts/` structure
- Cleanup unused hardhat dependencies and configs from `evm-contracts/`
- Altered Readme to explain current smart contract development setup
- Changed CI to run native compilation on both `evm-contracts/` and `contract/` directories
- Moved `check_solc_hashes` into `tools/ci/` as it now runs for both `evm-contracts/` and `contracts/`

### CODE REVIEW

**Review by se3000 — COMMENTED** [review]
How about `contracts/src` instead of `contracts/contracts`?

**Comment by alexroan** [comment]
> How about `contracts/src` instead of `contracts/contracts`?

Yep, that's what this change is 😁

**Comment by archseer** [comment]
I think you should also declare code owners for `contracts/`, it's currently only declared for `evm-contracts/`

**Review by samsondav — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -191,13 +193,20 @@ yarn setup
    yarn test
    ```
 
-   ii. Solidity versions `>=0.8.x`:
+#### Soldiity >=v0.8
```
[the remark this answers is not in the document]
Comment (se3000) [reply README.md:196]:
```suggestion
#### Solidity >=v0.8
```

**Review by se3000 — APPROVED** [review]

**Review by se3000 — APPROVED** [review]

**Pull request merged**